ci/gpu/build.sh

```diff
@@ -117,6 +117,49 @@ else
     gpuci_mamba_retry install cugraph pylibcugraph -c ${CONDA_BLD_DIR} -c ${CONDA_ARTIFACT_PATH}
 fi
 
+################################################################################
+# Identify relevant testsets to run in CI based on the ChangeList
+################################################################################
+
+fnames=()
+# Initialize all the run variables to true, as we want to run all the tests if the build_mode is NOT a pull-request
+run_cpp_tests="true" run_python_tests="true" run_nb_tests="true"
+if [ "$BUILD_MODE" == "pull-request" ]; then
+    PR_ENDPOINT="https://api.github.com/repos/rapidsai/cugraph/pulls/${PR_ID}/files"
+    fnames=(
+      $(
+      curl -s \
+      -X GET \
+      -H "Accept: application/vnd.github.v3+json" \
+      -H "Authorization: token $GHTK" \
+      "$PR_ENDPOINT" | \
+      jq -r '.[].filename'
+      )
+    )
+    # Initialize all the run variables to false, for pull-requests only, later, based on what's changed, these variables will be set to true
+    run_cpp_tests="false" run_python_tests="false" run_nb_tests="false"
+fi
+# this will not do anything if the 'fnames' array is empty
+for fname in "${fnames[@]}"
+do
+   if [[ "$fname" == *"cpp/cmake/"* || "$fname" == *"cpp/CMakeLists.txt"* || "$fname" == *"cpp/src/"* || "$fname" == *"cpp/include/"* || "$fname" == *"cpp/tests/"* || "$fname" == *"cpp/libcugraph_etl/"* || "$fname" == *"cpp/scripts/"* ]]; then
```

Comment:
I would personally err on the side of running the tests rather than skipping them when in doubt. If you enumerate every subdir and file that triggers testing, and we add a subdir or file that isn't listed, the tests won't get run.  By explicitly listing the cpp dir then only items to exclude, you don't risk missing test runs on newly added source dirs or files (for instance, in the past few months we added libcugraph_etl). Missing test runs could potentially go unnoticed for a while in that situation and lead to a broken upstream.

Reply:
You are correct Rick and I have update the commit based on your advise